File: 22FALL/240.search-a-2-d-matrix-ii.py
#
# @lc app=leetcode id=240 lang=python3
#
# [240] Search a 2D Matrix II
#

# @lc code=start
class Solution:
    def searchMatrix(self, matrix, target):
        m, n = len(matrix), len(matrix[0])
        i, j = 0, n-1
        while i < m and j >=0:
            if matrix[i][j] == target:
                return True
            elif matrix[i][j] < target:
                i += 1
            else:
                j -= 1
        return False

            

# @lc code=end

File: VSCode_CPP/240.search-a-2-d-matrix-ii.cpp
/*
 * @lc app=leetcode id=240 lang=cpp
 *
 * [240] Search a 2D Matrix II
 */

// @lc code=start
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int m=matrix.size(), n=matrix[0].size();
        int i=0, j=n-1;
        while(i<m && j>-1) {
            if(matrix[i][j]==target) return true;
            else if(matrix[i][j]>target) {
                if (j==0) return false;
                j--;
            } else if (matrix[i][j]<target) {
                if (i==m-1) return false;
                i++;
            }
        }
        return false;
    }
};
// @lc code=end

